frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,244 @@ pub struct PoolRoles<AccountId> {
+/// decreased after the initial value is set, to prevent commission from repeatedly increasing.
+///
+/// An optional commission `change_rate` allows the pool to set strict limits to how much commission
+/// can change in each update, and how often updates can take place. If a `change_rate` is set
+/// before a commission `current` is set, the initial commission `current` value will not be subject
+/// to throttling. Subsequent commission updates will be.
+#[derive(
+	Encode, Decode, DefaultNoBound, MaxEncodedLen, TypeInfo, DebugNoBound, PartialEq, Copy, Clone,
+)]
+#[codec(mel_bound(T: Config))]
+#[scale_info(skip_type_params(T))]
+pub struct Commission<T: Config> {
+	/// Optional commission rate of the pool along with the account commission is paid to.
+	pub current: Option<(Perbill, T::AccountId)>,
+	/// Optional maximum commission that can be set by the pool `root`. Once set, this value can
+	/// only be updated to a decreased value.
+	pub max: Option<Perbill>,
+	/// Optional configuration around how often commission can be updated, and when the last
+	/// commission update took place.
+	pub change_rate: Option<CommissionChangeRate<T::BlockNumber>>,
+	/// The block throttling should be checked from. This value will be updated on all commission
+	/// updates and when setting an initial `change_rate`.
+	pub throttle_from: Option<T::BlockNumber>,
+}
+
+impl<T: Config> Commission<T> {
+	/// Returns true if the current commission updating to `to` would exhaust the change rate
+	/// limits.
+	///
+	/// A commission update will be throttled (disallowed) if:
+	/// 1. not enough blocks have passed since the `throttle_from` block, if exists, or
+	/// 2. the new commission is greater than the maximum allowed increase.
+	fn throttling(&self, to: &Perbill) -> bool {
+		if let Some(t) = self.change_rate.as_ref() {
+			let commission_as_percent =
+				self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero());
+
+			// the attempted increase in commission relative to the current commission.
+			let attempted_increase = (*to).saturating_sub(commission_as_percent);
+
+			if self.throttle_from.map_or(
+				// Note: defensive only. `throttle_from` should always exist where `change_rate`
+				// has already been set, so this scenario should never happen.
+				//
+				// if no `throttle_from` exists. Therefore, the commission update is throttled via
+				// `max_increase` only.
+				attempted_increase > t.max_increase,
+				|f| {
+					// `min_delay` blocks must have been passed since `throttle_from`.
+					if <frame_system::Pallet<T>>::block_number().saturating_sub(f) < t.min_delay {
+						return true
+					}
+					// ensure the `max_increase` durations passed since the previous commission
+					// update allow the attempted commission increase.
+					//
+					// the total durations passed since the last commission update.
+					let blocks_passed = <frame_system::Pallet<T>>::block_number().saturating_sub(f);
```

Comment:
```suggestion
                                         // the total durations passed since the last commission update.
					let blocks_passed = <frame_system::Pallet<T>>::block_number().saturating_sub(f);
					if blocks_passed < t.min_delay {
						return true
					}
					
					
```